packages/ketcher-react/src/script/editor/tool/sgroup.ts

```diff
@@ -547,25 +554,27 @@ function getContextBySgroup(restruct, sgAtoms) {
   return anyChainedBonds(sgBonds) ? SgContexts.Group : SgContexts.Bond
 }
 
-function getContextBySelection(restruct, selection) {
+function getContextBySelection(restruct: ReStruct, selection: Selection) {
   const struct = restruct.molecule
 
   if (selection.atoms && !selection.bonds) {
     return SgContexts.Atom
   }
 
-  const bonds = selection.bonds.map((bondid) => struct.bonds.get(bondid))
-
+  const bonds = selection.bonds?.map((bondid) => struct.bonds.get(bondid))
   if (!anyChainedBonds(bonds)) {
     return SgContexts.Bond
   }
 
   selection.atoms = selection.atoms || []
 
   const atomSet = new Pile(selection.atoms)
-  const allBondsSelected = bonds.every(
-    (bond) => atomSet.has(bond.begin) && atomSet.has(bond.end)
-  )
+  const allBondsSelected = bonds?.every((bond) => {
+    if (bond) {
```

Comment:
Why is this check needed?

Reply:
apparently `bond` can be `undefined` and in this case bond.begin/end will throw an error.
Not sure how to better handle this, any hints?